### ISSUE TITLE
x-compress-hint and reproxy

### ISSUE THREAD

**Issue by utrenkner** [h2o/h2o#2013]
Somehow I cannot get my sitemap.xml compressed. The file is reproxied from a dynamically generated file - could this be the cause of my problem?

In my h2o.conf I have this
```
"/sitemap.xml":
  compress: ON
  reproxy: ON
  mruby.handler-file: /usr/local/etc/h2o/mruby/sitemap.rb
```

The handler under /usr/local/etc/h2o/mruby/sitemap.rb just reproxies to the actual, dynamically generated file:
```
lambda do |env|
  headers = {}
  reproxy_to = "/sitemap.html?format=xml"
  return [307, {"x-reproxy-url" => "#{reproxy_to}"}, []]
  [399, headers, []]
end
```

This reproxied-to file comes with the ``x-compress-hint`` header switched on, which also shows in h2o's reponse:
```
HTTP/2.0 200 OK
server: h2o/2.3.0-beta1
date: Wed, 20 Mar 2019 10:39:40 GMT
date: Wed, 20 Mar 2019 10:39:40 GMT
content-type: application/xml; charset=utf-8
x-compress-hint: on
strict-transport-security: max-age=31536000
x-xss-protection: 1; mode=block
x-content-type-options: nosniff
X-Firefox-Spdy: h2
```
The correct file is served but it is not compressed despite the request with ``Accept-Encoding: gzip, deflate, br``.

Does this mean that a reproxied file will never be compressed if it is not already?

**Comment by deweerdt** [comment]
`x-compress-hint` is a header that's understood by the proxy layer (`proxy.reverse.url`). That is, it only works if the header is present in the response that's received by a proxying layer.
That said, `compress: ON` should be enough to get a compressed object (regardless of the `x-compress-hint` header), maybe the object is too small to be compressed? There's a `compress-minimum-size` setting.

**Comment by utrenkner** [comment]
Thank you @deweerdt ! In the end, the problem was not the ``compress-minimum-size`` but the mime type. ``application/xml`` is not yet recognised by h2o as compressible. I changed the mime type to ``text/xml`` and now it gets compressed just fine.

Because ``application/xml`` is quite common, I created a [pull request to include it in the default is_compressible list](https://github.com/h2o/h2o/pull/2016).
